### ISSUE TITLE
step 7, npm start fails

### ISSUE THREAD

**Issue by sxsxsxin** [alexa/alexa-avs-sample-app#541]
pi@raspberrypi:~/Desktop/alexa-avs-sample-app/samples $ cd companionService && npm start

> alexa-voice-service-sample-companion-service@1.0.0 start /home/pi/Desktop/alexa-avs-sample-app/samples/companionService
> node ./bin/www

This node service needs to be running to store token information memory and vend them for the AVS app.

module.js:340
    throw err;
          ^
Error: Cannot find module 'express'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/pi/Desktop/alexa-avs-sample-app/samples/companionService/app.js:1:77)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)

npm ERR! alexa-voice-service-sample-companion-service@1.0.0 start: `node ./bin/www`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the alexa-voice-service-sample-companion-service@1.0.0 start script.
npm ERR! This is most likely a problem with the alexa-voice-service-sample-companion-service package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./bin/www
npm ERR! You can get their info via:
npm ERR!     npm owner ls alexa-voice-service-sample-companion-service
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 4.4.38-v7+
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "start"
npm ERR! cwd /home/pi/Desktop/alexa-avs-sample-app/samples/companionService
npm ERR! node -v v0.10.29
npm ERR! npm -v 1.4.21
npm ERR! code ELIFECYCLE
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian
 
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/pi/Desktop/alexa-avs-sample-app/samples/companionService/npm-debug.log
npm ERR! not ok code 0
pi@raspberrypi:

**Comment by sxsxsxin** [comment]
iĺl admit iḿ a bit out of my element here. the packages appeared to compile OK during the automated _install.sh.  Anyone have any ideas ?

**Comment by rinhbt** [comment]
same here, try to fix my npm and re-installed both nodejs and npm but still got stuck at the same problem.
Still waiting for help

**Comment by sanjayrd** [comment]
Looks like another user was having this issue. Does this thread help?
https://github.com/alexa/alexa-avs-sample-app/issues/530

**Comment by sxsxsxin** [comment]
I pulled the plug on the project after using a workaround to install all the missing modules referenced in the config.json , and then receiving a host of other SSL/Cert errors after the 3 modules were running. 
thanks anyway

**Comment by syncrasp** [comment]
I'm having the same issues, showing npm: command not found. Can you please advise on steps to resolve this?

Thx!

**Comment by ishc** [comment]
I had the same issue. npm was downloaded but not installed.
Try installing the npm before starting it:

npm install
npm start

**Comment by fortifieditsolutions** [comment]
ishc commented 12 days ago

I had the same issue. npm was downloaded but not installed.
Try installing the npm before starting it:

npm install
npm start

worked perfect for me. but just to clarify more:

pi@raspberrypi:~/Desktop/alexa-avs-sample-app/samples/companionService $ npm install
(it did its magic)
pi@raspberrypi:~/Desktop/alexa-avs-sample-app/samples/companionService $ npm start

> alexa-voice-service-sample-companion-service@1.0.0 start /home/pi/Desktop/alexa-avs-sample-app/samples/companionService
> node ./bin/www

This node service needs to be running to store token information memory and vend them for the AVS app.

Listening on port 3000

**Comment by elstryan** [comment]
Are any folks on this thread still running into this issue?

**Comment by krisnaw** [comment]
Hi @elstryan I have got this error when trying to run npm install
`npm WARN package.json node-storage@0.0.7 No repository field.
npm ERR! Failed to parse json
npm ERR! Unexpected token �
npm ERR! File: /home/pi/.npm/serve-static/1.9.3/package/package.json
npm ERR! Failed to parse package.json data.
npm ERR! package.json must be actual JSON, not just JavaScript.
npm ERR! 
npm ERR! This is not a bug in npm.
npm ERR! Tell the package author to fix their package.json file. JSON.parse

npm ERR! System Linux 4.4.50-v7+
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "install"
npm ERR! cwd /home/pi/Desktop/alexa-avs-sample-app/samples/companionService
npm ERR! node -v v0.10.29
npm ERR! npm -v 1.4.21
npm ERR! file /home/pi/.npm/serve-static/1.9.3/package/package.json
npm ERR! code EJSONPARSE
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/pi/Desktop/alexa-avs-sample-app/samples/companionService/npm-debug.log
npm ERR! not ok code 0
`

**Comment by yiyasha15** [comment]
if youre using pi3 then it'll show that error!
Type ( sudo apt-get install npm ) on the terminal and then try npm install, it'll work.

**Comment by yiyasha15** [comment]
@sxsxsxin Did you get a solution for the original problem?

**Comment by tweaty100** [comment]
hello @fortifieditsolutions  @sanjayrd  @rinhbt  @sxsxsxin  @ishanchho @yiyasha15 
how to resolve this npm warn package.json node-storage@0.0.7 no repository field in linux?please tell anyone facing same problem or have any prior knowledge will be appreciated.

Thanks

**Comment by ambelbasha** [comment]
I'm having the same problem. Here is my : **npm-debug.log** 

0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/nodejs', '/usr/bin/npm', 'start' ]
2 info using npm@1.4.21
3 info using node@v0.10.29
4 error Error: ENOENT, open '/home/pi/package.json'
5 error If you need help, you may report this *entire* log,
5 error including the npm and node versions, at:
5 error     <http://github.com/npm/npm/issues>
6 error System Linux 4.9.35-v7+
7 error command "/usr/bin/nodejs" "/usr/bin/npm" "start"
8 error cwd /home/pi
9 error node -v v0.10.29
10 error npm -v 1.4.21
11 error path /home/pi/package.json
12 error code ENOENT
13 error errno 34
14 verbose exit [ 34, true ]

**Comment by vaibhavbhanawat01** [comment]
have anyone found the solution of this issue ?? please let know if anyone find it??

**Comment by ambelbasha** [comment]
What I did and worked for me is simply run automated_installation.sh from the beginning , and keep in mind to enter every command line to a terminal by hand instead of using a bash script . The reason that I tried this is that I was seeing various errors in meantime that everything was in place and should work fine . Seems to me that every command through terminal needs their time to be executed and in order to get in line processes interacting with npm , java client or wake word . Would you try this out? This problems are frustrating and kept me spending time instead of doing other things that I should do. Hope I don't misdirected you .

**Comment by tocan** [comment]
I run Openplotter a solution for sailingboats and here the most important SignalK Server use port  3000 in Amazon developers you can choose yours port. I just need to know where in the install script i can change port 3000 to port 3100 for example...

Port 3000 is already in use
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! alexa-voice-service-sample-companion-service@1.0.0 start: `node ./bin/www`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the alexa-voice-service-sample-companion-service@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2017-09-16T14_15_39_583Z-debug.log

**Comment by PieterjanDeClippel** [comment]
Just to clarify:
to install npm:
`sudo apt-get -y install npm`

`npm install`
actually means something totally different.
This command installs/updates all node-modules which are required by a project you would have downloaded somewhere. Those dependencies are not included in some projects because your project would take a huge amount of space on eg. github or subversion, while these dependencies are duplicate-available on the internet.

In short: you create a project -> you need some packages from nodejs -> the packages are installed outside the project -> only the project files are versioned -> someone else clones the repo -> he needs to run `npm install` in the folder the packages.json resides -> the missing packages are installed.

**Comment by 2dareis2do** [comment]
running npm on my host machine seems to be one way around the problem
